### PR TITLE
fix:  enciphered data will Replaced by '******' when log output

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What type of this PR
/kind feature

#### What this PR does / why we need it:
fix:  enciphered data will Replaced by '******' when log output

#### Which issue(s) this PR fixes:
https://erda.cloud/erda/dop/projects/387/issues/all?id=206353&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIl0sImFzc2lnbmVlSURzIjpbIjEwMDEyNjEiXX0%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=506&type=BUG

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |       Desensitization of action agent log information       |
| 🇨🇳 中文    |      action agent 日志信息脱敏处理        |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sfwn** [comment]
`action` should be mentioned. Only action log will be handled.

File: modules/actionagent/file_watch.go [hunk 1]
```diff
@@ -83,8 +83,17 @@ func metaFileFullHandler(agent *Agent) filewatch.FullHandler {
 // stdoutTailHandler 以 tail 方式处理 stdout
 func stdoutTailHandler(agent *Agent) filewatch.TailHandler {
 	return func(line string, allLines []string) error {
-		// add your logic here
-		// logrus.Printf("stdout tailed a line: %s\n", line)
+		values := make([]string, 0)
```
Comment (sfwn) [inline modules/actionagent/file_watch.go:86]: abstract a method.

Comment (sfwn) [reply modules/actionagent/file_watch.go:86]: consider the order.

File: modules/actionagent/file_watch.go [hunk 2]
```diff
@@ -101,8 +110,17 @@ func stdoutTailHandler(agent *Agent) filewatch.TailHandler {
 // stderrTailHandler 以 tail 方式处理 stderr
 func stderrTailHandler(agent *Agent) filewatch.TailHandler {
 	return func(line string, allLines []string) error {
-		// add your logic here
-		// logrus.Printf("stderr tailed a line: %s\n", line)
+		values := make([]string, 0)
```
Comment (sfwn) [inline modules/actionagent/file_watch.go:113]: reuse method.

File: modules/actionagent/define.go [hunk 3]
```diff
@@ -62,6 +64,8 @@ type AgentArg struct {
 
 	PipelineID     uint64 `json:"pipelineID"`
 	PipelineTaskID uint64 `json:"pipelineTaskID"`
+
+	EncryptSecretKeys []string `json:"encryptSecretKeys"`
```
Comment (sfwn) [inline modules/actionagent/define.go:68]: Add comment.

File: modules/actionagent/pull_bootstrap_info.go [hunk 4]
```diff
@@ -27,6 +29,8 @@ import (
 	"github.com/erda-project/erda/pkg/retry"
 )
 
+const EncryptedValueLen = "ACTIONAGENT_ENCRYPTED_VAlUE_LEN"
```
Comment (sfwn) [inline modules/actionagent/pull_bootstrap_info.go:32]: min len

File: modules/pipeline/spec/pipeline_task.go [hunk 5]
```diff
@@ -95,6 +95,8 @@ type PipelineTaskExtra struct {
 	LoopOptions *apistructs.PipelineTaskLoopOptions `json:"loopOptions,omitempty"` // 开始执行后保证不为空
 
 	AppliedResources apistructs.PipelineAppliedResources `json:"appliedResources,omitempty"`
+
+	EncryptSecretKeys []string `json:"encryptSecretKeys"`
```
Comment (sfwn) [inline modules/pipeline/spec/pipeline_task.go:99]: no need. pipeline snapshot level is enough now.

File: modules/pipeline/spec/pipeline_extra.go [hunk 6]
```diff
@@ -145,7 +145,8 @@ type Snapshot struct {
 	BindQueue *apistructs.PipelineQueue `json:"bindQueue,omitempty"`
 
 	// Events stores pipeline level k8s-like events
-	Events []*apistructs.PipelineEvent `json:"events,omitempty"`
+	Events            []*apistructs.PipelineEvent `json:"events,omitempty"`
+	EncryptSecretKeys []string                    `json:"encryptSecretKeys"`
```
Comment (sfwn) [inline modules/pipeline/spec/pipeline_extra.go:149]: add a new line.

File: cmd/actionagent/main.go [hunk 7]
```diff
@@ -60,6 +60,7 @@ func realMain(args string) {
 	agent := &actionagent.Agent{
 		Errs:              make([]error, 0),
 		PushedMetaFileMap: make(map[string]string),
+		BlackList:         make([]string, 0),
```
Comment (sfwn) [inline cmd/actionagent/main.go:63]: black list for what?

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/actionagent/file_watch.go (hunk 1 above)
Comment (littlejiancc) [reply modules/actionagent/file_watch.go:86]: done

File: modules/actionagent/file_watch.go (hunk 2 above)
Comment (littlejiancc) [reply modules/actionagent/file_watch.go:113]: done

File: modules/actionagent/define.go (hunk 3 above)
Comment (littlejiancc) [reply modules/actionagent/define.go:68]: done

File: modules/actionagent/pull_bootstrap_info.go (hunk 4 above)
Comment (littlejiancc) [reply modules/actionagent/pull_bootstrap_info.go:32]: valueLen, err := strconv.Atoi(os.Getenv(EncryptedValueLen))
	if err != nil || valueLen == 0 {
		valueLen = 6
	}

File: modules/pipeline/spec/pipeline_extra.go (hunk 6 above)
Comment (littlejiancc) [reply modules/pipeline/spec/pipeline_extra.go:149]: done

File: cmd/actionagent/main.go (hunk 7 above)
Comment (littlejiancc) [reply cmd/actionagent/main.go:63]: done

File: modules/pipeline/spec/pipeline_task.go (hunk 5 above)
Comment (littlejiancc) [reply modules/pipeline/spec/pipeline_task.go:99]: the func `pullBootstrapInfo ` will use

File: cmd/actionagent/main.go [hunk 8]
```diff
@@ -60,6 +60,7 @@ func realMain(args string) {
 	agent := &actionagent.Agent{
 		Errs:              make([]error, 0),
 		PushedMetaFileMap: make(map[string]string),
+		BlackList:         make([]string, 0), // enciphered data will Replaced by '******' when log output
```
Comment (sfwn) [inline cmd/actionagent/main.go:63]: TextBlackList

File: modules/actionagent/file_watch.go [hunk 9]
```diff
@@ -120,6 +120,18 @@ func stderrTailHandler(agent *Agent) filewatch.TailHandler {
 	}
 }
 
+func replace(blackList []string, line *string) {
```
Comment (sfwn) [inline modules/actionagent/file_watch.go:123]: desensitizeLine

File: modules/actionagent/pull_bootstrap_info.go (hunk 4 above)
Comment (sfwn) [inline modules/actionagent/pull_bootstrap_info.go:32]: env name: ACTIONAGENT_ENCRYPTED_VAlUE_MIN_LEN

**Review by sfwn — CHANGES_REQUESTED** [review]

**Comment by sfwn** [comment]
Test min_len again and add screenshots.

Fulfill Code Patch for unit test.

File: modules/actionagent/pull_bootstrap_info.go (hunk 4 above)
Comment (littlejiancc) [reply modules/actionagent/pull_bootstrap_info.go:32]: done

File: modules/actionagent/file_watch.go (hunk 9 above)
Comment (littlejiancc) [reply modules/actionagent/file_watch.go:123]: done

File: modules/actionagent/file_watch.go [hunk 10]
```diff
@@ -83,8 +83,8 @@ func metaFileFullHandler(agent *Agent) filewatch.FullHandler {
 // stdoutTailHandler 以 tail 方式处理 stdout
 func stdoutTailHandler(agent *Agent) filewatch.TailHandler {
 	return func(line string, allLines []string) error {
-		// add your logic here
-		// logrus.Printf("stdout tailed a line: %s\n", line)
+		desensitizeLine(agent.TextBlackList, &line)
+		fmt.Printf("%s\n", line)
```
Comment (sfwn) [inline modules/actionagent/file_watch.go:87]: use fmt.Fprintf

File: modules/actionagent/file_watch.go [hunk 11]
```diff
@@ -101,8 +101,8 @@ func stdoutTailHandler(agent *Agent) filewatch.TailHandler {
 // stderrTailHandler 以 tail 方式处理 stderr
 func stderrTailHandler(agent *Agent) filewatch.TailHandler {
 	return func(line string, allLines []string) error {
-		// add your logic here
-		// logrus.Printf("stderr tailed a line: %s\n", line)
+		desensitizeLine(agent.TextBlackList, &line)
+		fmt.Errorf("%s\n", line)
```
Comment (sfwn) [inline modules/actionagent/file_watch.go:105]: fmt.Fprintf

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/actionagent/file_watch.go (hunk 10 above)
Comment (littlejiancc) [reply modules/actionagent/file_watch.go]: done

File: modules/actionagent/file_watch.go (hunk 11 above)
Comment (littlejiancc) [reply modules/actionagent/file_watch.go]: done

**Review by sfwn — APPROVED** [review]

**Comment by sfwn** [comment]
add unit test for patch.

File: modules/pipeline/services/pipelinesvc/secret.go [hunk 12]
```diff
@@ -217,9 +217,12 @@ func (s *PipelineSvc) FetchSecrets(p *spec.Pipeline) (secrets, cmsDiceFiles map[
 				GlobalDecrypt:  true,
 			})
 		if err != nil {
-			return nil, nil, nil, err
+			return nil, nil, nil, nil, err
 		}
 		for _, c := range configs.Data {
+			if c.EncryptInDB && c.Type == "kv" {
```
Comment (sfwn) [inline modules/pipeline/services/pipelinesvc/secret.go:223]: use cms.ConfigTypeKV, not plain text.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/services/pipelinesvc/secret.go (hunk 12 above)
Comment (littlejiancc) [reply modules/pipeline/services/pipelinesvc/secret.go:223]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**